lib/matplotlib/figure.py

```diff
@@ -1133,11 +1120,25 @@ def draw(self, renderer):
         dsu = [row for row in dsu if not row[1].get_animated()]
         dsu.sort(key=itemgetter(0))
 
-        mimage._draw_list_compositing_images(
-            renderer, self, dsu, self.suppressComposite)
+        try:
+            renderer.open_group('figure')
+            # prevent triggering call backs during the draw process
```

Comment:
actually, on a bit further consideration that line does not matter.  In the first pass at the stale system which did not propagate 'stale' state past an already stale state, however during the development (to support live updates in a plain python prompt) we made the 'staleness' always propogate all the way up (which may be related to performance issues in https://github.com/matplotlib/matplotlib/issues/6664) so that you could put a call back on the stale transition to draw idle (in IPython we use their 'post-execute' hook to kick the drawing)